### DISCUSSION TITLE
Testing modal visibility in jest/react-testing-library

### DISCUSSION THREAD

**Discussion by taitems** [chakra-ui/chakra-ui#5796] [Q&A]
I am attempting to convert an existing `antd` modal to a Chakra Modal. This is straight forward, but I am struggling with the existing test. I've tried the `renderWithPortal` method but to no success.

Code sandbox here: https://codesandbox.io/s/dank-cherry-0lpxi8?file=/components/lorem.tsx:301-1005

Sample component:

```js
<>
    <Text
    as="a"
    color="purple.500"
    onClick={onOpen}
    data-testid="trigger-me-in-jest"
    >
    Open me
    </Text>
    <Modal isOpen={isOpen} onClose={onClose}>
    <ModalOverlay />
    <ModalContent>
        <ModalHeader>Modal Title</ModalHeader>
        <ModalCloseButton />
        <ModalBody>
        <div data-testid="find-me-in-jest">Hi!</div>
        </ModalBody>

        <ModalFooter>
        <Button colorScheme="blue" mr={3} onClick={onClose}>
            Close
        </Button>
        <Button variant="ghost">Secondary Action</Button>
        </ModalFooter>
    </ModalContent>
    </Modal>
</>
```

Sample test
```js
describe("Accordion", () => {
  it("renders correctly", async () => {
    render(<Lorem />);

    const trigger = screen.getByTestId("trigger-me-in-jest");

    fireEvent.click(trigger);

    expect(screen.getByTestId("find-me-in-jest")).toBeVisible();
  });
});
```

**Comment by joeprivettsmart** [comment]
I might be wrong, but changing the open/closed state of the Chakra Modal does not change the style attributes on the modal in a way that is detected by the `toBeVisible` matcher. I believe it changes the `transform` styling, e.g. 

For invisibility:
`expect(screen.getByTestId("find-me-in-jest")).toHaveStyle('transform: translateX(0px) translateY(0.18967%) translateZ(0);')`

**Comment by rjmccluskey** [comment]
I spent way too long trying to figure out why my test was failing before finding this issue... Thanks @joeprivettsmart!

**Comment by dmtrs** [comment]
Had similar issue that resolved with `waitFor`. I think this is related with the animation... here is a pseudo code:
```ts
import { waitFor } from '@testing-library/react';
// ... other imports here

it('should close dialog', async() => {
  // Arrange
  // ...
  const open = screen.getByRole('button', { name: 'Open modal' });
  // Act
  await user.click(open);
  // Assert
  expect(screen.getByRole('dialog')).toBeVisible();
  
  // 2nd Act
  await user.click(screen.getByRole('button', { name: 'Close' }));
  
  // Assert
  await waitFor(() => expect(screen.getByRole('dialog')).not.toBeVisible());
});
```
